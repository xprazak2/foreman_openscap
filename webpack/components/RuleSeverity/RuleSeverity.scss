img.severity-img-scale {
  height: 30%;
}
