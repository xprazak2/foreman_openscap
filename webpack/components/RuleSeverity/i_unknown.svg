<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 19.2.1, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 110 130" style="enable-background:new 0 0 110 130;" xml:space="preserve">
<style type="text/css">
	.st0{fill:#CCCCCC;}
	.st1{fill:#FFFFFF;}
</style>
<path class="st0" d="M96.6,130H13c-7.2,0-13-5.9-13-13l0,0c0-7.2,5.9-13,13-13h83.6c7.2,0,13,5.9,13,13l0,0
	C109.7,124.1,103.8,130,96.6,130z M96.6,95.4H13c-7.2,0-13-5.9-13-13l0,0c0-7.2,5.9-13,13-13h83.6c7.2,0,13,5.9,13,13l0,0
	C109.7,89.5,103.8,95.4,96.6,95.4z M97,60.2H13.4c-7.2,0-13-5.9-13-13l0,0c0-7.2,5.9-13,13-13H97c7.2,0,13,5.9,13,13l0,0
	C110,54.4,104.1,60.2,97,60.2z M96.6,26.1H13c-7.2,0-13-5.9-13-13l0,0C0,5.9,5.9,0,13,0h83.6c7.2,0,13,5.9,13,13l0,0
	C109.7,20.2,103.8,26.1,96.6,26.1z"/>
<path class="st0" d="M53.3,123.4c-4.6,0-8.5-1.6-11.6-4.6c-3.2-3.1-4.8-7-4.8-11.5s1.6-8.4,4.8-11.5c1-1,2.1-1.8,3.3-2.5h-4.9v-8.7
	c0-4.9,0.7-9.3,2.1-13.3c1.4-4,3.3-7.3,5.5-9.8c1.9-2.1,4.1-4.3,6.3-6.3c2.2-2,4.3-3.8,6.2-5.3c1.4-1.1,2.5-2.5,3.5-4.2
	c0.9-1.6,1.3-3.2,1.3-5c0-2.7-0.8-4.7-2.5-6.3c-1.7-1.7-4.2-2.5-7.6-2.5c-4.2,0-7.3,1.2-9.4,3.5c-2.3,2.5-3.4,6.1-3.4,10.9V51H15.6
	v-4.5c0-12.2,3.7-22,10.9-29.2c7.2-7.1,16.7-10.8,28.4-10.8c11.3,0,20.3,3.2,26.8,9.6c6.5,6.4,9.8,14.7,9.8,24.7
	c0,4.4-0.7,8.5-2.2,12.1c-1.5,3.6-3.4,6.6-5.6,8.9c-1.9,1.9-3.9,3.8-6.1,5.9c-2.1,2-4.1,3.8-6,5.5c-1.4,1.2-2.6,2.9-3.6,4.9
	c-1,2-1.4,4.2-1.4,6.7v8.5h-4.8c1.2,0.7,2.3,1.6,3.4,2.6c3.2,3.2,4.7,7,4.7,11.4c0,4.5-1.6,8.4-4.8,11.5
	C61.8,121.9,57.9,123.4,53.3,123.4z"/>
<path class="st1" d="M54.8,11.1c10.1,0,18,2.8,23.6,8.3c5.6,5.6,8.4,12.7,8.4,21.5c0,3.9-0.6,7.4-1.9,10.4c-1.2,3.1-2.8,5.5-4.6,7.4
	c-1.8,1.8-3.8,3.8-6,5.7c-2.1,2-4.1,3.8-6,5.4c-1.8,1.6-3.4,3.8-4.6,6.3c-1.2,2.6-1.9,5.5-1.9,8.6v4H44.6v-4.2
	c0-4.4,0.6-8.3,1.9-11.8c1.2-3.5,2.8-6.2,4.6-8.3c1.8-2,3.8-4,6-6c2.1-1.9,4.1-3.6,6-5.1c1.8-1.4,3.4-3.3,4.6-5.5
	c1.2-2.2,1.9-4.6,1.9-7.2c0-3.9-1.3-7.1-3.9-9.5c-2.6-2.5-6.2-3.7-10.7-3.7c-5.5,0-9.7,1.7-12.7,5c-3,3.3-4.5,8-4.5,13.9H20.1
	c0-11,3.2-19.7,9.5-26C36,14.2,44.4,11.1,54.8,11.1 M53.3,95.7c3.4,0,6.2,1.1,8.5,3.4c2.3,2.3,3.4,5,3.4,8.2c0,3.3-1.1,6-3.4,8.3
	c-2.3,2.2-5.1,3.4-8.5,3.4c-3.4,0-6.2-1.1-8.5-3.4c-2.3-2.2-3.4-5-3.4-8.3c0-3.3,1.1-6,3.4-8.3C47.1,96.8,50,95.7,53.3,95.7
	 M54.8,2.1C42,2.1,31.4,6.1,23.3,14.1c-8.1,8-12.2,18.9-12.2,32.4v9h9h17.4h9v-9c0-3.6,0.7-6.3,2.2-7.9c0.8-0.9,2.3-2.1,6.1-2.1
	c2.1,0,3.7,0.4,4.5,1.2c0.5,0.5,1.1,1.2,1.1,3c0,1.1-0.2,2-0.7,2.9c-0.7,1.2-1.4,2.1-2.3,2.8c-2,1.6-4.2,3.4-6.5,5.5
	c-2.3,2.1-4.6,4.3-6.6,6.6c-2.6,2.9-4.8,6.7-6.4,11.3c-1.6,4.4-2.4,9.4-2.4,14.8v4.2v7.5c-2.1,3.2-3.1,7-3.1,11
	c0,5.7,2.1,10.8,6.1,14.7c4,3.9,9.1,5.9,14.8,5.9c5.7,0,10.8-2,14.8-5.9c4-3.9,6.1-9,6.1-14.7c0-4.1-1.1-7.9-3.3-11.1v-7.3v-4
	c0-1.8,0.3-3.4,1-4.7c0.7-1.5,1.6-2.7,2.5-3.5c1.9-1.7,4-3.6,6.1-5.6c2.2-2,4.3-4,6.2-6c2.7-2.7,4.9-6.2,6.6-10.4
	c1.7-4.1,2.5-8.8,2.5-13.8c0-11.2-3.7-20.6-11.1-27.8C77.5,5.7,67.4,2.1,54.8,2.1L54.8,2.1z"/>
</svg>
