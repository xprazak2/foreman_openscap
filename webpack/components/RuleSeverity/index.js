import React from 'react';

import SeverityCritical from './i_severity-critical.svg';
import SeverityHigh from './i_severity-high.svg';
import SeverityMedium from './i_severity-med.svg';
import SeverityLow from './i_severity-low.svg';
import SeverityUnknown from './i_unknown.svg';

import './RuleSeverity.scss';

const RuleSeverity = props => {
  const imgStyles = { className: 'severity-img-scale' };

  const propsMapping = {
    'Low': { alt: 'Low Serverity', src: SeverityLow, ...imgStyles },
    'Medium': { alt: 'Medium Serverity', src: SeverityMedium, ...imgStyles },
    'High': { alt: 'High Serverity', src: SeverityHigh, ...imgStyles },
    'Critical': { alt: 'Critical Serverity', src: SeverityCritical, ...imgStyles },
    'Unknown': { alt: 'Unknown Serverity', src: SeverityUnknown, ...imgStyles },
  };

  let imgProps = propsMapping[props.severity] || propsMapping['Unknown'];
  return <img {...imgProps} />;
}

export default RuleSeverity;
