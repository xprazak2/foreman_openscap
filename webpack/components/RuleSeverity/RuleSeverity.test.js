import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import RuleSeverity from './index';

const levels = ['Low', 'Medium', 'High', 'Critical', 'foo'];

const fixtures = levels.reduce((memo, level) => {
  memo[`should render for ${level} severity`] = { severity: level };
  return memo;
}, {});

describe('RuleSeverity', () =>
  testComponentSnapshotsWithFixtures(RuleSeverity, fixtures));
