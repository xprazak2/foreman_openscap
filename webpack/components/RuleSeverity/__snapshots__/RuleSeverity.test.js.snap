// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuleSeverity should render for Critical severity 1`] = `
<img
  alt="Critical Serverity"
  className="severity-img-scale"
  src={[Function]}
/>
`;

exports[`RuleSeverity should render for High severity 1`] = `
<img
  alt="High Serverity"
  className="severity-img-scale"
  src={[Function]}
/>
`;

exports[`RuleSeverity should render for Low severity 1`] = `
<img
  alt="Low Serverity"
  className="severity-img-scale"
  src={[Function]}
/>
`;

exports[`RuleSeverity should render for Medium severity 1`] = `
<img
  alt="Medium Serverity"
  className="severity-img-scale"
  src={[Function]}
/>
`;

exports[`RuleSeverity should render for foo severity 1`] = `
<img
  alt="Unknown Serverity"
  className="severity-img-scale"
  src={[Function]}
/>
`;
